test/integration/local_lambda/test_end_to_end.py

```diff
@@ -5,227 +5,234 @@
-                Popen(cmd.split(" ")).communicate()
-            
-        for arch in ARCHS:
-            arch_tag = "" if arch == "" else f"-{arch}"
-            Popen(f"docker rmi {cls.image_name}{arch_tag}".split(" ")).communicate()
+        Popen(f"docker rmi {cls.image_name}".split(" ")).communicate()
 
-    def tagged_name(self, name, architecture):
-        tag = self.get_tag(architecture)
-        return (name + tag, "aws-lambda-rie" + tag, self.image_name + tag)
+    def tagged_name(self, name):
+        tag = self.get_tag()
+        return (name + tag, "aws-lambda-rie" + tag, self.image_name)
 
-    def get_tag(self, architecture):
-        return "" if architecture == "" else str(f"-{architecture}")
+    def get_tag(self):
+        return "" if self.ARCH == "" else str(f"-{self.ARCH}")
     
     def run_command(self, cmd):
         Popen(cmd.split(" ")).communicate()
     
     def sleep_1s(self):
         time.sleep(SLEEP_TIME)
         
-    def invoke_function(self, port):
+    def invoke_function(self):
         return requests.post(
-            f"http://localhost:{port}/2015-03-31/functions/function/invocations", json={}
+            f"http://localhost:{self.PORT}/2015-03-31/functions/function/invocations", json={}
         )
         
-    def create_container_and_invoke_function(self, cmd, port):
-        self.run_command(cmd)
+    @contextmanager
+    def create_container(self, param, image):
+        try:
+            platform = "x86_64" if self.ARCH == "" else self.ARCH
+            cmd_full = f"docker run --platform linux/{platform} {param}"
+            self.run_command(cmd_full)
         
-        # sleep 1s to give enough time for the endpoint to be up to curl
-        self.sleep_1s()
+            # sleep 1s to give enough time for the endpoint to be up to curl
+            self.sleep_1s()
+            yield
+        except Exception as e:
+            print(f"An error occurred while executing cmd: {cmd_full}. error: {e}")
+            raise e
+        finally:
+            self.run_command(f"docker stop {image}")
+            self.run_command(f"docker rm -f {image}")
         
-        return self.invoke_function(port)
 
-    @parameterized.expand([("x86_64", "8000"), ("arm64", "9000"), ("", "9050")])
-    def test_env_var_with_equal_sign(self, arch, port):
-        image, rie, image_name = self.tagged_name("envvarcheck", arch)
-
-        cmd = f"docker run --name {image} -d -v {self.path_to_binary}:/local-lambda-runtime-server -p {port}:8080 --entrypoint /local-lambda-runtime-server/{rie} {image_name} {DEFAULT_1P_ENTRYPOINT} main.check_env_var_handler"
+    def test_env_var_with_equal_sign(self):
+        image, rie, image_name = self.tagged_name("envvarcheck")
+        cmd = f"--name {image} -d -v {self.path_to_binary}:/local-lambda-runtime-server -p {self.PORT}:8080 --entrypoint /local-lambda-runtime-server/{rie} {image_name} {DEFAULT_1P_ENTRYPOINT} main.check_env_var_handler"
```

Comment:
nit: change this variable to `param` or `params` in each test to match its actual usage

Reply:
You changed it in every other test except in this one 😄

Reply:
oh my bad